src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpAuthHelper.cs

```diff
@@ -171,10 +171,23 @@ public void PreAuthenticateRequest(WinHttpRequestState state, uint proxyAuthSche
             // 407-401-407-401- loop.
             if (proxyAuthScheme != 0)
             {
+                ICredentials proxyCredentials;
+                Uri proxyUri;
+                if (state.Proxy != null)
+                {
+                    proxyCredentials = state.Proxy.Credentials;
+                    proxyUri = state.Proxy.GetProxy(state.RequestMessage.RequestUri);
+                }
+                else
+                {
+                    proxyCredentials = state.DefaultProxyCredentials;
+                    proxyUri = state.RequestMessage.RequestUri;
```

Comment:
Since the else case is exactly what it is today, should I just leave all of this but still open an issue to track doing the right fix for the else case?